src/lib/utils/api/index-api.ts

```diff
@@ -18,4 +18,19 @@ export class IndexApi {
       throw error
     }
   }
+
+  async put(path: string, data: Record<string, unknown>) {
+    try {
+      const resp = await fetch(`${IndexApiBaseUrl}${path}`, {
+        method: 'PUT',
```

Comment:
Should probably check if would always just want one record or need a history of records when the user signed e.g. updated terms.